### PR TITLE
[quant][graphmode] Support quantization for `aten::append`

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39645 [quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing
* **#39644 [quant][graphmode] Support quantization for `aten::append`**
* #39615 [quant][graphmode] Fix a corner case in handling `if` in insert_observers

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942112](https://our.internmc.facebook.com/intern/diff/D21942112)

### CODE REVIEW

File: torch/csrc/jit/passes/quantization/helper.cpp [hunk 1]
```diff
@@ -283,6 +283,9 @@ std::vector<Value*> getPassThroughInputs(Value* v) {
       inputs.push_back(v);
     }
     return inputs;
+  } else if (n->kind() == Symbol::aten("append")) {
```
Comment (raghuramank100) [inline torch/csrc/jit/passes/quantization/helper.cpp:286]: How about adding a simple test to check for append?

**Review by raghuramank100 — APPROVED** [review]
